src/main/java/org/springframework/data/repository/aot/generate/AotQueryMethodGenerationContext.java

```diff
@@ -127,6 +127,16 @@ public TypeName getReturnTypeName() {
 		return TypeName.get(getReturnType().getType());
 	}
 
+	/**
+	 * Suggest naming clash free variant for the given intended variable name within the local method context.
+	 *
+	 * @param variableName the intended variable name.
+	 * @return the suggested VariableName
+	 */
+	public String suggestLocalVariableName(String variableName) {
```

Comment:
I would suggest refining the way we approach the problem. We generally hold details in the context and can look these up. I would suggest that we keep a map around with parameter name mappings and use this method as `getLocalVariableName(…)` or `localVariable(String logicalName)`. One could then call the method with the same name twice. On the first call, we would ensure there is no clash, the second call would return the mapped name. Would also alleviate the need to track the variable name in the calling code.

Reply:
sorry I don't get that comment. Why would I want to call the same method twice? 
you'll have to keep track of variable name in the calling code since you might have to apply it in different code blocks or have to pass it around between query creation and execution so you refer to right thing.

Reply:
> you'll have to keep track of variable name in the calling code since you might have to apply it in different code blocks or have to pass it around between query creation and execution so you refer to right thing.

Calling code would then not be required to track the actual variable name but could refer to the context so the context provides the information.